### PR TITLE
Comment likes

### PR DESCRIPTION
**Pull request by tgomezzzz**
This PR adds likes to each comment.

### CODE REVIEW

File: portfolio/src/main/webapp/index.html [hunk 1]
```diff
@@ -113,14 +113,22 @@ <h3>Gallery</h3>
...
-          <input type="text" id="author" name="author"><br></br>
-          <label for="message">Message:</label><br>
-          <input type="text" id="message" name="message"></br></br>
-          <input type="submit" id="submit"/>
-        </form>
-        <input type="number" id="max-comments" name="max-comments" min="0" value="5" onchange="fetchComments(value)"></br></br>
+        <div id="input-container">
+          <div id="comment-input">
+            <form action="/data" method="POST">
+              <label for="author-input">Name:</label></br>
```
Comment (vidhyal) [inline portfolio/src/main/webapp/index.html]: Renaming such variable is not required for the purpose of this particular PR. If you think this is required, then please do so in a separate PR, explaining the rationale behind renaming.

File: portfolio/src/main/java/com/google/sps/servlets/LikeCommentServlet.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    Entity comment = null;
+    try {
+      comment = datastore.get(KeyFactory.stringToKey(commentKeyString));
+    } catch (EntityNotFoundException e) {
+      e.printStackTrace();
+      response.sendRedirect("/");
+      return;
+    }
+
+    // The first element in this list records whether or not the client has like this comment, and the second records the number of likes this comment has.
```
Comment (vidhyal) [inline portfolio/src/main/java/com/google/sps/servlets/LikeCommentServlet.java]:
```suggestion
    // The first element in this list records whether or not the client has liked this comment, and the second records the number of likes this comment has.
```

File: portfolio/src/main/java/com/google/sps/servlets/LikeCommentServlet.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    try {
+      comment = datastore.get(KeyFactory.stringToKey(commentKeyString));
+    } catch (EntityNotFoundException e) {
+      e.printStackTrace();
+      response.sendRedirect("/");
+      return;
+    }
+
+    String clientIp = request.getRemoteAddr();
+    Set<String> likedByIpSet;
```
Comment (vidhyal) [inline portfolio/src/main/java/com/google/sps/servlets/LikeCommentServlet.java:86]: It might be more efficient to let comment have the property likedBy always and use the number of elements in likedByIpSet to determine the course of action.

File: portfolio/src/main/java/com/google/sps/servlets/LikeCommentServlet.java [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+    // If the comment has already been liked at least once, add or remove this client's IP to the set of IPs that have liked it.
+    if (comment.hasProperty("likedBy")) {
```
Comment (vidhyal) [inline portfolio/src/main/java/com/google/sps/servlets/LikeCommentServlet.java:89]: May be this property can be renamed to "likes" and likedByIpSet renamed to "likedBy"

**Review by adamtcohen — APPROVED** [review]
As discussed only did brief for review for critical issues.
